### DISCUSSION TITLE
More custom autocommands

### DISCUSSION THREAD

**Discussion by TimUntersberger** [NeogitOrg/neogit#1634] [Ideas]
Add more custom autocommands similiar to #69.

A few events that come to my mind:

* NeogitCommit
* NeogitCommitAbort
* NeogitBeforeCommit
* ...

**Comment by RianFuro** [comment]
Maybe we could use the `operations.lua`-thingy to manage that centrally. I was planning to also hook that into a channel-structure (from [here](https://github.com/nvim-lua/plenary.nvim/blob/b6eb19f560dd39212a43b26f4e88a69f16da8617/lua/plenary/async_lib/util.lua#L154)), so consumers from the lua side can be notified about changes. (if that's how it works, haven't looked into that too much)

**Comment by halfdan** [comment]
I would love to see `NeogitPullComplete` to automatically trigger recompilation of the project I'm working on.

**Comment by halfdan** [comment]
Pushed a PR to add that one event. 

Unsure about using operations.lua - the file mentions it's but a stop-gap solution to deal with plenary shortcomings. Didn't want to build on top of something that was due to be replaced.

Maybe @TimUntersberger can advise here.
